security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,37 @@
+package crypter
+
+import (
+	"errors"
+)
+
+// counterLen is the byte length of the counter.
+const counterLen = 8
+
+// counter is a 64-bit, little-endian counter.
+type counter struct {
+	value         uint64
+	hasOverflowed bool
+}
+
+func newCounter(value uint64) counter {
+	return counter{value: value}
+}
+
+// getAndIncrement returns the current value of the counter and increments it.
+func (c *counter) getAndIncrement() (uint64, error) {
+	if c.hasOverflowed {
+		return 0, errors.New("invalid counter due to overflow")
```

Comment:
Could we change this to `counter has overflowed`? We no longer have a concept of an invalid counter.

Reply:
Done